### DISCUSSION TITLE
Need a bit of help about using the ALS Refactored

### DISCUSSION THREAD

**Discussion by saravanan290702** [Sixze/ALS-Refactored#492] [Q&A]
Hello I am relatively new to unreal engine and had been trying to make locomotion systems none of them were good. Then i recently started looking at the als in marketplace, and later came to know about the als refactored. 
I know every little C++, but i wanted to know if i can add more features on  the als refactored using blueprints. Will i be able to implement them? 
I just need to add a swimming, melee combat and a simple climb to the als refactored using blueprints. how do i make it possible any advices..

**Comment by ameaninglessname** [comment]
There should be videos about how to make `swimming` or `melee` in UnrealEngine on `YouTube`. You could learn these first and then think about how to integrate them into ALS.

**Reply by saravanan290702** [reply]
Will i be able to add them over the als refactored using bp?

**Reply by MrWilsonWilson** [reply]
Yes you will be, i have added swimming in our game and plugged it directly into the ALS refactored system, there is even a comment in the ANIMBP made by the developers as to where you should place it
